api/src/neo4j/queries/map.js

```diff
@@ -1,8 +1,40 @@
 import querySingleResult from 'neo4j/queryHandlers/single';
 import queryListResult from 'neo4j/queryHandlers/list';
 import parseParams from 'neo4j/shared/helper';
+const fs = require('fs');
+const readline = require('readline');
 
-const getMapsListing = async ({ model, version }) => {
+const mapReactionIdSet = async (map, modelShortName) => {
+  const rl = readline.createInterface({
+      input: fs.createReadStream(`/project/svg/${modelShortName}/${map.filename}`),
+      output: process.stdout,
+      terminal: false
+  });
+
+  const mapReactionIdSet = new Set();
+  const getReactions = async () =>{
+    for await (const line of rl) {
+      if (line.includes('class="rea"')) {
+        const match = line.match(/id="(.+)"\s/);
+        if (match) {
+          mapReactionIdSet.add(match[1]);
+        }
+      }
+    }
+  }
+  await getReactions()
```

Comment:
This was my bad, I think I used this nested function during prototyping but realize now that the `getReactions` functions is not needed. The following should work:

```suggestion
  const mapReactionIdSet = new Set();
  for await (const line of rl) {
    if (line.includes('class="rea"')) {
      const match = line.match(/id="(.+)"\s/);
      if (match) {
        mapReactionIdSet.add(match[1]);
      }
    }
  }
```